src/services/lens.ts

```diff
@@ -60,7 +46,8 @@ export class LensService<T extends ChainId> extends ContractService<T> {
    * @returns list of registry addresses
    */
   async getAdapters(overrides: CallOverrides = {}): Promise<string[]> {
-    return await this.contract.read.getRegistries(overrides);
+    let contract = await this.contract;
+    return contract.read.getRegistries(overrides);
```

Comment:
Wondering the reasoning for this change, perhaps better to just

```suggestion
    return this.contract.read.getRegistries(overrides);
```

as `return await` is redundant

Reply:
don't we still need this since `this.contract` is a promise now? Also should it be a `const` instead of `let`?

Reply:
Thing is , as per the request [here](https://github.com/yearn/yearn-sdk/pull/208#discussion_r816841312), the lens service address is not included in the AddressProvider, so this probably needs a revert, what do you think @jstashh ?

Reply:
> don't we still need this since `this.contract` is a promise now? Also should it be a `const` instead of `let`?

No because the function is `async`, and if you look at the return type is a `Promise<string[]>`, so the `await` is irrelevant in this case. Unless I'm missing something :)

Reply:
Totally right guys, my bad, didn't realize that `this.contract` is now a promise 🤦